UnityProject/Assets/Scripts/Objects/Doors/Modules/CrowbarModule.cs

```diff
@@ -11,177 +11,162 @@ public class CrowbarModule : DoorModuleBase
 		{
 			return;
 		}
 
-		private void TryPry(HandApply interaction)
+		private void PryDoor(HandApply interaction, bool useTool, bool isClosing)
 		{
-			if (master.IsClosed && !master.IsPerformingAction)
+
+			if (useTool == true)
 			{
-				if (master.TryForceOpen())
+				if(Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.CanPryDoor))
 				{
-					Chat.AddActionMsgToChat(interaction.Performer,
-						$"You force the {doorName} open with your {interaction.HandObject.ExpensiveName()}!",
-						$"{interaction.Performer.ExpensiveName()} forces the {doorName} open!");
+					master.SoundController.ServerPlaySound(DoorSoundController.DoorSoundType.JawsPry);
 				}
 				else
 				{
-					Chat.AddActionMsgToChat(interaction.Performer, $"The {doorName} does not budge at all!",
-						$"{interaction.Performer.ExpensiveName()} tries to force the {doorName} open, and fails!");
+					master.SoundController.ServerPlaySound(DoorSoundController.DoorSoundType.ToolPry);
 				}
+				
+
+				ToolUtils.ServerUseToolWithActionMessages(interaction, pryTime,
+					$"You start {(isClosing ? "forcing" : "prying")} the {doorName} {(isClosing ? "closed" : "open")}...",
+					$"{interaction.Performer.ExpensiveName()} starts {(isClosing ? "forcing" : "prying")} the {doorName} {(isClosing ? "closed" : "open")}...",
+					$"",
+					$"",
+					() => TryPry(interaction, useTool, isClosing), onFailComplete: OnFailPry, playSound: false);
 			}
-			else if (!master.IsClosed && !master.IsPerformingAction)
+
+			else if (useTool == false)
 			{
-				if ((crowbarRequiresNoPower && master.HasPower) &&
-				    (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.CanPryDoor) == false))
-				{
-					return;
-				}
+				master.SoundController.ServerPlaySound(DoorSoundController.DoorSoundType.HandPry);
+
+				Chat.AddActionMsgToChat(interaction.Performer,
+					$"You start {(isClosing ? "forcing" : "prying")} the {doorName} {(isClosing ? "closed" : "open")}...",
+					$"{interaction.Performer.ExpensiveName()} starts {(isClosing ? "forcing" : "prying")} the {doorName} {(isClosing ? "closed" : "open")} with its {interaction.PerformerPlayerScript.PlayerTypeSettings.PryHandName}...");
+
+				var cfg = new StandardProgressActionConfig(StandardProgressActionType.Construction);
 
-				master.Close();
+				StandardProgressAction.Create(
+					cfg,
+					() => TryPry(interaction, useTool, isClosing)
+				).ServerStartProgress(master.RegisterTile, pryTime, interaction.Performer);
 			}
 		}
 
-		private void TryPryHand(HandApply interaction)
+		private void TryPry(HandApply interaction, bool useTool, bool isClosing)
 		{
-			if (master.IsClosed == false && master.IsPerformingAction == false)
+			//Refuse if door is in motion
+			if (master.IsPerformingAction) return;
+
+			//Refuse if its a crowbar and the door has power
+			else if ((crowbarRequiresNoPower && master.HasPower) &&
```

Comment:
At the end of each of these if's you return. So they dont actually need to be an else-if chain as the further code won't execute anyways. Codacy notes this as redundant jumps.